### PR TITLE
Emby.Server.Implementations code cleanup and warning fixes (Part 1)

### PR DESCRIPTION
**Pull request by barronpm**
**Changes**
Documents and cleans up code in Emby.Server.Implementations.

This is the first part of a series of PR's that hopefully will eliminate all warnings in the Emby.Server.Implementations project.

**Related Issues**
#2149

### CODE REVIEW

File: Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs [hunk 1]
```diff
@@ -27,6 +25,10 @@
 
 namespace Emby.Server.Implementations.Activity
 {
+    /// <summary>
+    /// The activity log entry point.
+    /// <see cref="IServerEntryPoint"/>.
```
Comment (Bond-009) [inline Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs:30]:
```suggestion
```
No need, you can see the implemented interfaces in the api docs

File: Emby.Server.Implementations/Activity/ActivityManager.cs [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
 using System;
 using MediaBrowser.Controller.Library;
 using MediaBrowser.Model.Activity;
 using MediaBrowser.Model.Events;
 using MediaBrowser.Model.Querying;
-using Microsoft.Extensions.Logging;
 
 namespace Emby.Server.Implementations.Activity
 {
+    /// <inheritdoc />
```
Comment (Bond-009) [inline Emby.Server.Implementations/Activity/ActivityManager.cs:9]: Should have it's own documentation

File: Emby.Server.Implementations/Activity/ActivityRepository.cs [hunk 3]
```diff
@@ -15,18 +13,28 @@
 
 namespace Emby.Server.Implementations.Activity
 {
+    /// <inheritdoc cref="IActivityRepository" />
```
Comment (Bond-009) [inline Emby.Server.Implementations/Activity/ActivityRepository.cs:16]: Should have it's own documentation

File: Emby.Server.Implementations/Channels/ChannelDynamicMediaSourceProvider.cs [hunk 4]
```diff
@@ -27,12 +28,9 @@ public ChannelDynamicMediaSourceProvider(IChannelManager channelManager)
...
         {
-            if (item.SourceType == SourceType.Channel)
-            {
-                return _channelManager.GetDynamicMediaSources(item, cancellationToken);
-            }
-
-            return Task.FromResult<IEnumerable<MediaSourceInfo>>(new List<MediaSourceInfo>());
+            return item.SourceType == SourceType.Channel
+                ? _channelManager.GetDynamicMediaSources(item, cancellationToken)
+                : Task.FromResult<IEnumerable<MediaSourceInfo>>(new List<MediaSourceInfo>());
```
Comment (Bond-009) [inline Emby.Server.Implementations/Channels/ChannelDynamicMediaSourceProvider.cs:33]:
```suggestion
                : Task.FromResult<IEnumerable<MediaSourceInfo>>(Enumerable.Empty<MediaSourceInfo>());
```
You'll have to add `using System.Linq;` at the top of the file

File: Emby.Server.Implementations/Collections/CollectionManager.cs [hunk 5]
```diff
@@ -23,6 +21,7 @@
 
 namespace Emby.Server.Implementations.Collections
 {
+    /// <inheritdoc />
```
Comment (Bond-009) [inline Emby.Server.Implementations/Collections/CollectionManager.cs:24]: Shouldn't inherit doc

**Comment by barronpm** [comment]
@Bond-009 Whenever a class `Something` is implementing `ISomething`, shouldn't the documentation be done at the interface level? The docs would be the same for both, so in order to avoid documenting twice, the implementing class should inherit from the interface, no?

**Comment by Bond-009** [comment]
Yes, it *can* be the same, but it *shouldn't*. Ideally the class should have more information

File: Emby.Server.Implementations/Channels/ChannelManager.cs [hunk 6]
```diff
@@ -919,59 +961,55 @@ private BaseItem GetChannelItemEntity(ChannelItemInfo info, IChannel channelProv
...
+                        break;
+                    default:
+                        if (info.ContentType == ChannelMediaContentType.Trailer || info.ExtraType == ExtraType.Trailer)
+                        {
+                            item = GetItemById<Trailer>(info.Id, channelProvider.Name, out isNew);
+                        }
+                        else
+                        {
+                            item = GetItemById<Video>(info.Id, channelProvider.Name, out isNew);
+                        }
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Channels/ChannelManager.cs:1010]: This default block is kind of strange looking, I wonder if the original version was more readable.

I think you could also turn this into a switch expression. This should would work:

```cs
                item = info.ContentType switch
                {

                    ChannelMediaContentType.Episode => GetItemById<Episode>(info.Id, channelProvider.Name, out isNew),
                    ChannelMediaContentType.Movie => GetItemById<Movie>(info.Id, channelProvider.Name, out isNew),
                    var x when x == ChannelMediaContentType.Trailer || info.ExtraType == ExtraType.Trailer
                        => GetItemById<Trailer>(info.Id, channelProvider.Name, out isNew),
                    _ => GetItemById<Video>(info.Id, channelProvider.Name, out isNew)
                };
```

**Pull request merged**